src/chat_module.ts

```diff
@@ -506,24 +512,76 @@ export class ChatModule implements ChatInterface {
           throw new Error("Last messages should be a string from the `user`.");
         }
         this.getPipeline().appendConversationMessage(
-          message.name ? message.name : roles[0],
+          Role.User,
           message.content,
         );
+        if (message.name !== undefined) {
+          this.getPipeline().overrideUserRoleName(message.name);
+        }
       } else if (message.role === "assistant") {
         if (typeof message.content !== "string") {
           // TODO(Charlie): Remove when we support function calling
```

Comment:
```suggestion
          // Prevent `null` message history; even for function calling we expect a `string` in our implementation.
```

I added this check before because the doc for `content` says _"The contents of the assistant message. Required unless `tool_calls` is specified"_. 

This is mainly for the case when user wants to provide a history of function callings. I guess in our implementation `content` is never null, so this check should still be preserved.

We should also write a note about our implementation in the doc for `ChatCompletionAssistantMessageParam` and `ChatCompletionMessage`, specifically that we currently do not populate `tool_calls` but only use `content` instead.